tests/test_pyglmnet.py

```diff
@@ -345,3 +345,42 @@ def test_cdfast():
 def test_fetch_datasets():
     """Test fetching datasets."""
     datasets.fetch_community_crime_data('/tmp/glm-tools')
+
+def test_random_state_consistency():
+    """ Test model's random_state """
+
+    # Generate the dataset
+    n_samples, n_features = 1000, 100
+
+    np.random.seed(42)
+
+    beta0 = 1. / (np.float(n_features) + 1.) * \
+            np.random.normal(0.0, 1.0)
+    beta = 1. / (np.float(n_features) + 1.) * \
+           np.random.normal(0.0, 1.0, (n_features,))
+    Xtrain = np.random.normal(0.0, 1.0, [n_samples, n_features])
+
+
+    # Test the consistency for each solver
+    distrs = ['softplus', 'gaussian', 'poisson', 'binomial', 'probit']
+    solvers = ['batch-gradient', 'cdfast']
+
+    for solver in solvers:
+        for distr in distrs:
+
+            ytrain = simulate_glm(distr, beta0, beta, Xtrain,
+                                   sample=False, random_state=42)
+
+            glm = GLM(distr=distr, solver=solver, alpha=0.01, reg_lambda=0.1, random_state=1)
+            glm.fit(Xtrain,ytrain)
+            result_a = glm.predict(Xtrain)
+
+            # Train again the same model
+            glm = GLM(distr=distr, solver=solver, alpha=0.01, reg_lambda=0.1, random_state=1)
```

Comment:
can you edit an existing test rather than adding a completely new test?

Reply:
I would be slightly more in favour to keep this new test since it avoids to have only one big test that checks for all the possible cases (I would be happy to do a refactoring of tests in order to reduce copy-paste code). However, if you think it would be better to move this test into another one, I will update it accordingly.

Reply:
I see what you mean. But I think it's overkill to test all the solvers and distributions for this. It will increase the testing time. We just need to run a `fit_predict` twice on a GLM object (and maybe also GLMCV?) and compare their results. So, it shouldn't be more than 3-4 lines of code

Reply:
I updated the test to run only with `fit_predict`, without checking for all the possible solvers/distributions combinations. You are also right about `GLMCV`, so I've added it as well.
fortunately, this test case does not take so long to run in the end.